File: go/3202. Find the Maximum Length of Valid Subsequence II.go
func maximumLength(nums []int, k int) int {
	answer := 2
	for i := 0; i < k; i++ {
		freq := make([]int, k)
		count := 0
		for _, v := range nums {
			v %= k
			freq[v] = freq[(i-v+k)%k] + 1
			count = max(count, freq[v])
		}
		answer = max(answer, count)
	}
	return answer
}

File: java/3202. Find the Maximum Length of Valid Subsequence II.java
class Solution {
    public int maximumLength(int[] nums, int k) {
        int answer = 2;
        for (int i = 0; i < k; i++) {
            int[] freq = new int[k];
            int count = 0;
            for (int v : nums) {
                v %= k;
                freq[v] = freq[(i-v+k)%k]+1;
                count = Math.max(count, freq[v]);
            }
            answer = Math.max(answer, count);
        }
        return answer;
    }
}